storage/remote/queue_manager_test.go

```diff
@@ -2332,24 +2332,147 @@ func BenchmarkBuildTimeSeries(b *testing.B) {
 	}
 }
 
-func TestPopulateV2TimeSeries_UnexpectedMetadata(t *testing.T) {
+func TestPopulateV2TimeSeries_typeAndUnitLabels(t *testing.T) {
 	symbolTable := writev2.NewSymbolTable()
-	pendingData := make([]writev2.TimeSeries, 4)
 
-	batch := []timeSeries{
-		{sType: tSample, seriesLabels: labels.FromStrings("__name__", "metric1")},
-		{sType: tMetadata, seriesLabels: labels.FromStrings("__name__", "metric2")},
-		{sType: tSample, seriesLabels: labels.FromStrings("__name__", "metric3")},
-		{sType: tMetadata, seriesLabels: labels.FromStrings("__name__", "metric4")},
+	testCases := []struct {
+		name                    string
+		typeLabel               string
+		unitLabel               string
+		expectedType            writev2.Metadata_MetricType
+		description             string
+		enableTypeAndUnitLabels bool
```

Comment:
Do we need this variable if we always set it to true?

Another problem is that we used to test the unexpected metadata regression here (so no metadata and enableTypeAndUnitLabels = false case), but now we don't test it anymore, right?